minidns-dnssec/src/main/java/de/measite/minidns/dnssec/algorithms/AlgorithmMap.java

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright 2015 the original author or authors
+ *
+ * This software is licensed under the Apache License, Version 2.0,
+ * the GNU Lesser General Public License version 2 or later ("LGPL")
+ * and the WTFPL.
+ * You may choose either license to govern your use of this software only
+ * upon the condition that you accept all of the terms of either
+ * the Apache License 2.0, the LGPL 2.1+ or the WTFPL.
+ */
+package de.measite.minidns.dnssec.algorithms;
+
+import de.measite.minidns.dnssec.DNSSECValidatorInitializationException;
+import de.measite.minidns.dnssec.DigestCalculator;
+import de.measite.minidns.dnssec.SignatureVerifier;
+import de.measite.minidns.record.NSEC3;
+
+import java.security.NoSuchAlgorithmException;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import static de.measite.minidns.DNSSECConstants.DIGEST_ALGORITHM_SHA1;
+import static de.measite.minidns.DNSSECConstants.DIGEST_ALGORITHM_SHA256;
+import static de.measite.minidns.DNSSECConstants.SIGNATURE_ALGORITHM_DSA;
+import static de.measite.minidns.DNSSECConstants.SIGNATURE_ALGORITHM_DSA_NSEC3_SHA1;
+import static de.measite.minidns.DNSSECConstants.SIGNATURE_ALGORITHM_ECC_GOST;
+import static de.measite.minidns.DNSSECConstants.SIGNATURE_ALGORITHM_ECDSAP256SHA256;
+import static de.measite.minidns.DNSSECConstants.SIGNATURE_ALGORITHM_ECDSAP384SHA384;
+import static de.measite.minidns.DNSSECConstants.SIGNATURE_ALGORITHM_RSAMD5;
+import static de.measite.minidns.DNSSECConstants.SIGNATURE_ALGORITHM_RSASHA1;
+import static de.measite.minidns.DNSSECConstants.SIGNATURE_ALGORITHM_RSASHA1_NSEC3_SHA1;
+import static de.measite.minidns.DNSSECConstants.SIGNATURE_ALGORITHM_RSASHA256;
+import static de.measite.minidns.DNSSECConstants.SIGNATURE_ALGORITHM_RSASHA512;
+
+public class AlgorithmMap {
+    private Logger LOGGER = Logger.getLogger(AlgorithmMap.class.getName());
+
+    private Map<Byte, DigestCalculator> dsDigestMap;
+    private Map<Byte, SignatureVerifier> signatureMap;
+    private Map<Byte, DigestCalculator> nsecDigestMap;
+
+    @SuppressWarnings("deprecation")
+    public AlgorithmMap() {
+        dsDigestMap = new ConcurrentHashMap<>();
+        nsecDigestMap = new ConcurrentHashMap<>();
```

Comment:
ConcurrentHashMap is overkill. Use a local variable, fill that one and wrap it with Collections.unmodifyableMap before asignment to the instance variable (Calling methods on an instance while the constructor has not yet finished is unsafe anyway).